common/djangoapps/student/models.py

```diff
@@ -1589,3 +1589,11 @@ def user_can_skip_entrance_exam(cls, user, course_key):
             except EntranceExamConfiguration.DoesNotExist:
                 can_skip = False
         return can_skip
+
+
+class LanguageProficiency(models.Model):
+    """
+    Represents a user's language proficiency.
+    """
+    user_profile = models.ForeignKey(UserProfile, db_index=True, related_name='language_proficiencies')
```

Comment:
I don't think it makes much of a difference performance-wise which model we attach to, and it does seem profile-y info, so that works for me. I believe doing a foreign key in MySQL means an index is made implicitly anyway, but declaring it is also fine. As for code... are we going to allow people to search across users and filter by language code? If so, sure.